src/Psr17Factory.php

```diff
@@ -0,0 +1,278 @@
+<?php
+
+namespace Http\Discovery;
+
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestFactoryInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UploadedFileFactoryInterface;
+use Psr\Http\Message\UploadedFileInterface;
+use Psr\Http\Message\UriFactoryInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * A portable PSR-17 implementation.
```

Comment:
could we call this a convenience class that provides all the various psr17 factories in one class?

Reply:
I added a note about this in the docblock of the class

Reply:
It a bit more than that: all known PRS-17 implementations provide all methods in one class, and this one is no different.
I changed the comment to "A generic PSR-17 implementation.", since generic is really what this is about: some independent from the actual implementation.